boards: lilygo: ttgo_tbeam: initial support

lfelten opened the pull request:
adds initial support for Lilygo TTGO TBeam board

Depends on #82474 to be merged.

lfelten committed:
boards: lilygo: ttgo_tbeam: initial support

adds initial support for Lilygo TTGO TBeam board

Co-authored-by: Jamie <40387179+nordicjm@users.noreply.github.com>
Co-authored-by: Benjamin Cabé <benjamin@zephyrproject.org>

Signed-off-by: Lothar Felten <lothar.felten@gmail.com>

nordicjm commented on boards/lilygo/ttgo_tbeam/ttgo_tbeam_esp32_procpu_defconfig:
```diff
@@ -0,0 +1,16 @@
+CONFIG_UART_INTERRUPT_DRIVEN=y
+
+CONFIG_GPIO=y
+CONFIG_I2C=y
+CONFIG_REGULATOR=y
+CONFIG_GNSS=y
+CONFIG_GNSS_SATELLITES=y
+CONFIG_GNSS_NMEA_GENERIC=y
+CONFIG_MFD=y
```
```suggestion
```

Unless hello_world needs satellite tracking or whatnot, these need to go

lfelten replied:
Hi nordicjm, 
in the documentation, I list the following samples:

* :zephyr:code-sample:`gnss`
* :zephyr:code-sample:`character-frame-buffer`

These will not work anymore if I remove I2C and GNSS.

I can remove REGULATOR and MFD if needed, but then the PMIC voltages might be set to odd defaults.

gmarull replied:
Then you need to fix your sample to select the appropriate Kconfig options.

nordicjm replied:
MFD/REGULATOR can stay

nordicjm commented on boards/lilygo/ttgo_tbeam/ttgo_tbeam-pinctrl.dtsi:
```diff
@@ -0,0 +1,56 @@
+			output-high;
+		};
+	};
+
```
```suggestion
```

nordicjm commented on boards/lilygo/ttgo_tbeam/ttgo_tbeam-pinctrl.dtsi:
```diff
@@ -0,0 +1,56 @@
+#include <zephyr/dt-bindings/pinctrl/esp32-gpio-sigmap.h>
+
+&pinctrl {
+
```
```suggestion
```

lfelten commented on boards/lilygo/ttgo_tbeam/ttgo_tbeam_esp32_procpu_defconfig:
```diff
@@ -0,0 +1,16 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_MAIN_STACK_SIZE=2048
```
@nordicjm This is set by many ESP32-based boards, it's also set by some STM32 based boards, one aspeed board and some qemu targets.
Many architectures set a default in the arch/<architecture>/Kconfig (arc,arm64,riscv,x86,mips,sparc) but arm does not set it there.
Please let me know what you recommend.

nordicjm replied:
remove it, there should be no setting of stack sizes in defconfig files

lfelten commented:
Depends on  PR 87139
https://github.com/zephyrproject-rtos/zephyr/pull/87139

lfelten commented:
@aasinclair @gmarull @danieldegrasse @kartben 
Just rebased to the current main.
Should I rebase to main and push periodically?

kartben commented:
> @aasinclair @gmarull @danieldegrasse @kartben Just rebased to the current main. Should I rebase to main and push periodically?

no need as long as CI passes -- however, can you look at addressing @nordicjm's comments when you have a chance?

lfelten commented:
@aasinclair @gmarull @danieldegrasse @kartben 
Just rebased to the current main.
Should I rebase to main and push periodically?

> > @aasinclair @gmarull @danieldegrasse @kartben Just rebased to the current main. Should I rebase to main and push periodically?
> 
> no need as long as CI passes -- however, can you look at addressing @nordicjm's comments when you have a chance?

Hi @kartben I fixed the ttgo_tbeam-pinctrl.dtsi (whitespace), ttgo_tbeam_esp32_procpu.dts (whitespace) and ttgo_tbeam_esp32_procpu_defconfig (remove CONFIG_MAIN_STACK_SIZE) a while ago.
Comparing ttgo_tbeam* to ttgo_lora32* I might now have to fix ttgo_lora32-pinctrl.dtsi.

nordicjm commented on boards/lilygo/ttgo_tbeam/ttgo_tbeam-pinctrl.dtsi:
```diff
@@ -0,0 +1,54 @@
+			pinmux = <UART0_TX_GPIO1>;
+			output-high;
+		};
+		group2 {
```
newline gap between nodes is missing

lfelten replied:
@nordicjm Hi Jamie,
sorry, I don't get it. Should there be a newline before group2?

lfelten replied:
Ok, a newline has been inserted before all "group2" nodes.

nordicjm commented on boards/lilygo/ttgo_tbeam/ttgo_tbeam_esp32_procpu.dts:
```diff
@@ -0,0 +1,158 @@
+
+	leds {
+		compatible = "gpio-leds";
+		red_led: led_0 {
```
and here

nordicjm replied:
Rules were introduced after the older boards were added, nrf54h20 is newer and should comply. There is no tool. Rules are on https://docs.zephyrproject.org/latest/contribute/style/devicetree.html

lfelten replied:
I added an empty line before each node, please review

nordicjm commented on boards/lilygo/ttgo_tbeam/ttgo_tbeam_esp32_procpu.dts:
```diff
@@ -0,0 +1,160 @@
+	scl-gpios = <&gpio0 22 GPIO_OPEN_DRAIN>;
+	pinctrl-0 = <&i2c0_default>;
+	pinctrl-names = "default";
+	ssd1306_128x64: ssd1306@3c {
```
not resolved, do not mark as resolved if it's not fixed
